### PR TITLE
Fix issue: systemctl daemon-reload would sporadically cause udev handler fail

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
#### Why I did it

A workaround to back port the fix for a systemd issue.

The systemd issue: https://github.com/systemd/systemd/issues/24668
The systemd PR to fix the issue: https://github.com/systemd/systemd/pull/24673/files

The formal solution should upgrade systemd to a version that contains the fix. But, systemd is a very basic service, upgrading systemd requires heavy test. 

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

Copy the correct systemd-udevd.service file in build time 

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [x] 202211

#### Tested branch (Please provide the tested image version)

- [x] 202211
- [ ] 

```
SONiC Software Version: SONiC.fix-udev.3-b65c7bdec_Internal
SONiC OS Version: 11
Distribution: Debian 11.7
Kernel: 5.10.0-18-2-amd64
Build commit: b65c7bdec
Build date: Mon Jun 19 10:54:50 UTC 2023
Built by: sw-r2d2-bot@r-build-sonic-ci02-241

Platform: x86_64-mlnx_msn4700-r0
HwSKU: ACS-MSN4700
ASIC: mellanox
ASIC Count: 1
Serial Number: MT2022X08597
Model Number: MSN4700-WS2FO
Hardware Revision: A1
Uptime: 08:10:11 up 1 min,  1 user,  load average: 1.81, 0.67, 0.24
Date: Sun 25 Jun 2023 08:10:11

Docker images:
REPOSITORY                    TAG                             IMAGE ID       SIZE
docker-fpm-frr                fix-udev.3-b65c7bdec_Internal   a7b911e7cb6f   346MB
docker-fpm-frr                latest                          a7b911e7cb6f   346MB
docker-platform-monitor       fix-udev.3-b65c7bdec_Internal   94c5178cf80b   731MB
docker-platform-monitor       latest                          94c5178cf80b   731MB
docker-orchagent              fix-udev.3-b65c7bdec_Internal   46b393e0ace8   328MB
docker-orchagent              latest                          46b393e0ace8   328MB
docker-syncd-mlnx             fix-udev.3-b65c7bdec_Internal   1f5c6c23e33a   734MB
docker-syncd-mlnx             latest                          1f5c6c23e33a   734MB
docker-sflow                  fix-udev.3-b65c7bdec_Internal   7e45992c8c59   317MB
docker-sflow                  latest                          7e45992c8c59   317MB
docker-teamd                  fix-udev.3-b65c7bdec_Internal   e4d905592cda   316MB
docker-teamd                  latest                          e4d905592cda   316MB
docker-nat                    fix-udev.3-b65c7bdec_Internal   7fe799367580   319MB
docker-nat                    latest                          7fe799367580   319MB
docker-macsec                 latest                          d702a5554171   318MB
docker-snmp                   fix-udev.3-b65c7bdec_Internal   3bce8fcf71cd   338MB
docker-snmp                   latest                          3bce8fcf71cd   338MB
docker-sonic-telemetry        fix-udev.3-b65c7bdec_Internal   f13949cbc817   597MB
docker-sonic-telemetry        latest                          f13949cbc817   597MB
docker-dhcp-relay             latest                          153d9072805d   306MB
docker-router-advertiser      fix-udev.3-b65c7bdec_Internal   aed642b9a6bc   299MB
docker-router-advertiser      latest                          aed642b9a6bc   299MB
docker-sonic-p4rt             fix-udev.3-b65c7bdec_Internal   a3cae5ca65a7   870MB
docker-sonic-p4rt             latest                          a3cae5ca65a7   870MB
docker-mux                    fix-udev.3-b65c7bdec_Internal   b81f0401b9a8   347MB
docker-mux                    latest                          b81f0401b9a8   347MB
docker-eventd                 fix-udev.3-b65c7bdec_Internal   c5917d0e801f   298MB
docker-eventd                 latest                          c5917d0e801f   298MB
docker-lldp                   fix-udev.3-b65c7bdec_Internal   fd5dc14a7976   341MB
docker-lldp                   latest                          fd5dc14a7976   341MB
docker-database               fix-udev.3-b65c7bdec_Internal   438c2715a1dd   299MB
docker-database               latest                          438c2715a1dd   299MB
docker-sonic-mgmt-framework   fix-udev.3-b65c7bdec_Internal   5c50b115fbcd   414MB
docker-sonic-mgmt-framework   latest  
```

### CODE REVIEW

**Review by keboliu — APPROVED** [review]

**Comment by nazariig** [comment]
@Junchao-Mellanox what's the benefit of copying an entire file? Can we use scripts to just remove the offending part?

**Comment by Junchao-Mellanox** [comment]
> @Junchao-Mellanox what's the benefit of copying an entire file? Can we use scripts to just remove the offending part?

Hi @nazariig , copying entire file is simple, clear and robust.

**Comment by qiluo-msft** [comment]
As PR template, Please provide the tested image version.

**Comment by Junchao-Mellanox** [comment]
> As PR template, Please provide the tested image version.

Updated

File: build_debian.sh [hunk 1]
```diff
@@ -447,6 +447,9 @@ sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y in
     systemd-sysv \
     ntp
 
+# Workaround to fix issue https://github.com/systemd/systemd/issues/24668.
+sudo cp files/image_config/systemd/systemd-udevd/systemd-udevd.service $FILESYSTEM_ROOT/lib/systemd/system/systemd-udevd.service
```
Comment (qiluo-msft) [inline build_debian.sh:451]: You need to apply a patch to the systemd source code, and build a deb package.

**Comment by qiluo-msft** [comment]
Is it a real issue in SONiC context?
Patching systemd also requires heavy test. How do you verify?

**Comment by qiluo-msft** [comment]
Please provide the tested image version: The image version is part of `show version`, it is okay to have hash in the version string.

**Comment by Junchao-Mellanox** [comment]
It is a real issue. I ran a long term test like more than 24 hours to do two actions in parallel:
1. config reload
2. systemctl daemon-reload (running in a while loop)

I also run sonic-mgmt regression test on different platforms

File: build_debian.sh (hunk 1 above)
Comment (Junchao-Mellanox) [reply build_debian.sh:451]: yes, this is the best way to do it. But building systemd might take long and need heavy test like you mentioned. How about we set a systemd hash that contain the fix?

Comment (Junchao-Mellanox) [reply build_debian.sh:451]: Hi @qiluo-msft , I just checked, we may use bullseye-backports. What do you think?

Comment (qiluo-msft) [reply build_debian.sh:451]: Use bullseye-backports in a bullseye host is an acceptable solution.

Comment (qiluo-msft) [reply build_debian.sh:451]: Any update on bullseye-backports?

**Comment by qiluo-msft** [comment]
Please provide the tested image version: The image version is part of `show version`, it is okay to have hash in the version string.

File: build_debian.sh (hunk 1 above)
Comment (Junchao-Mellanox) [reply build_debian.sh]: I have built an image. Now I am running regression on that image.

File: files/build_templates/sonic_debian_extension.j2 [hunk 2]
```diff
@@ -228,12 +228,14 @@ sudo cp -f $IMAGE_CONFIGS/bash/bash.bashrc $FILESYSTEM_ROOT/etc/
 # Install readline's initialization file
 sudo cp -f $IMAGE_CONFIGS/readline/inputrc $FILESYSTEM_ROOT/etc/
 
+sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y install libsystemd-dev -t bullseye-backports
```
Comment (nazariig) [inline files/build_templates/sonic_debian_extension.j2:231]: @Junchao-Mellanox do we need to have `libsystemd0` also?

Comment (Junchao-Mellanox) [reply files/build_templates/sonic_debian_extension.j2:231]: i will check this

**Comment by Junchao-Mellanox** [comment]
Hi, just an update. After installing bullseye-backports of udev, it solves the issue. But, it introduces another issue: https://github.com/systemd/systemd/issues/25811. So, we might need to build systemd from source and backports the fix manually.

**Comment by Junchao-Mellanox** [comment]
Hi @qiluo-msft , I have finished the test. Could you please help review it?

**Review by qiluo-msft — APPROVED** [review]

**Comment by Junchao-Mellanox** [comment]
Hi @StormLiangMS , could you please help cherry-pick to 202211?

**Comment by StormLiangMS** [comment]
@Junchao-Mellanox is this an issue also for 202305 and 202205?

**Pull request merged**